proposals/0041-i18n-routing.md

```diff
@@ -0,0 +1,377 @@
+- Start Date: 2023/10/19
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+First-class support for localized routes, aka i18n routing.
+
+
+# Background & Motivation
+
+Many websites need to ship support for translated/localised websites for many reasons:
+- legal;
+- localised content;
+- localised market;
+- etc.
+
+Nowadays, there are [workarounds](https://docs.astro.build/en/recipes/i18n/) in Astro to make it work, although these workarounds have limitations, and because of that many users can't ship their website properly, or they have to work more to compensate the lack of first-class support.
+
+# Goals
+
+- Localised routes with locale prefixes;
+- Default locale with no prefix;
+- Redirect to default locale if prefix enabled;
+- Localise injected routes;
+- Domain support, with the help of [Adapter features](https://docs.astro.build/en/reference/adapter-reference/#adapter-features), so this will be bound to the limitations of the hosting provider;
+- Provide the necessary APIs for integrations and libraries to request information about the current locales;
+- Locale detection via the `Accept-Language` header, so support SSR;
+- Provide first-class APIs to users to work around locales (`.astro` components, endpoints, middleware);
+
+# Non-Goals
+
+- Localised data (dates, numbers, plurals, et.c);
+- Dictionaries where users can store translations of pre-defined words;
+- SEO optimisations;
+
+This gives the reader the correct context on what is intentionally left out of scope.
+It is okay to leave this section empty.
+
+# Detailed Design
+
+## Terminology
+
+- Locale: a code that represents a language
+- Localized folder/directory: a folder/directory named after a locale
+
+## Opt-in configuration
+
+The feature is opt-in, to avoid disrupting existing websites. To enable the feature, a new configuration called `i18n` is available:
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config"
+export default defineConfig({
+    i18n: {
+        
+    }
+})
+```
+
+The feature requires **two required** fields, where the user needs to store the default locale via `defaultLocale`, and a list of available locales via `locales`:
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config"
+export default defineConfig({
+    i18n: {
+        defaultLocaLe: 'en',
+        locales: ['en', 'es', 'pt', 'fr']
+    }
+})
+```
+
+Astro will throw an error if the `defaultLocale` value is not present in the list of `locales`.
+
+## File system based
+
+The localized directories must inside the `pages/` directory. Other than this restriction, the user is free to place the localized folders anywhere. 
```

Comment:
I don't think I understand this answer, it doesn't really matter, right? The implementation doesn't know about / care about where the underlying pages are coming from, just that they answer with a `200`?

Reply:
No, the implementation doesn't care about it, we care remove this paragraph if you think it can mislead.